docs/guides/basic-rbac.mdx

```diff
@@ -3,267 +3,303 @@ title: Implement basic Role Based Access Control (RBAC) with metadata
 description: Learn how to leverage Clerk's publicMetadata to implement your own basic Role Based Access Controls.
 ---
 
-To control which users can access certain parts of your application, you can leverage Clerk's [roles](/docs/organizations/roles-permissions#roles) feature. Although Clerk offers a roles feature as part of the feature set for [organizations](/docs/organizations/overview), not every app implements organizations. This guide will cover a workaround for setting up a basic Role Based Access Control (RBAC) system for products that don't use Clerk's organizations or roles.
+To control which users can access certain parts of your application, you can leverage Clerk's [roles](/docs/organizations/roles-permissions#roles) feature. Although Clerk offers a roles feature as part of the feature set for [organizations](/docs/organizations/overview), not every app implements organizations. **This guide covers a workaround to set up a basic Role Based Access Control (RBAC) system for products that don't use Clerk's organizations or roles.**
 
-This guide assumes that you are using Next.js App Router. The concepts here can be adapted to Next.js Pages Router and Remix.
+This guide assumes that you're using Next.js App Router, but the concepts can be adapted to any React-based framework.
 
-## Configure the session token
+<Steps>
+  ### Configure the session token
 
-Clerk provides [user metadata](/docs/users/metadata#user-metadata), which is a tool that can be leveraged to build flexible custom logic into your application. Metadata can be used to store information, and in this case, it can be used to store a user's role.
+  Clerk provides [user metadata](/docs/users/metadata#user-metadata), which can be used to store information, and in this case, it can be used to store a user's role.
 
-`unsafeMetadata` can be read and updated in the browser, and because the user could modify this metadata it should be treated as unsafe and validated by your application before trusting it. `privateMetadata` can not be read or modified in the browser. `publicMetadata` can be read by the browser and can only be updated server-side or in the Clerk Dashboard, making it the safest and best choice for this use case.
+  There are three types of metadata:
 
-To build a basic RBAC system, first, you need to make `publicMetadata` available to the application directly from the session token. With `publicMetadata` attached directly to the user's session, a fetch and network request isn't required every time you need to access the data.
+  - `unsafeMetadata` - can be both read and modified in the browser, meaning users can modify this data. Therefore, you should validate `unsafeMetadata` before trusting it for sensitive operations.
+  - `privateMetadata` - cannot be accessed nor modified in the browser.
+  - `publicMetadata` - can be accessed in the browser, but only be modified server-side or in the Clerk Dashboard.
```

Comment:
The revisions on this section have been frequent. Maybe we don't repeat this and instead link to the metadata section? That said, this list is accurate.

Reply:
[cf6613c](https://github.com/clerk/clerk-docs/pull/1560/commits/cf6613ce0a01e144d691d4098372d1867080757e)